File: atividade-02/python/arq.py
import os

def main():
    # cria um novo objeto File para ser o arquivo .txt
    file_path = "arq-novo.txt"
    
    # cria um array para separar cada linha do arquivo em strings
    lines = []

    # Condicional que fará o scanner ler todo o arquivo até que não haja nenhuma linha e que adiciona toda linha no array criado acima
    with open(file_path, 'r') as file:
        lines = file.readlines()
    
    # remove a última linha do array porque ela é uma string vazia
    if lines[-1] == '\n':
        lines.pop()

    # Separa a primeira linha do arquivo, contida no array "lines" e transforma a string em um array, usando o comando split para separar cada elemento e o separando em outro array, formando assim o array definido pelo arquivo .txt lido
    numbers = lines[0].strip().split(" ")
    definitive_list = numbers
    lines.pop(0)
    
    instruction_list = []

    for line in lines:
        instructions = line.strip().split(" ")
        instruction_list.append(instructions)

    for instruction in instruction_list:
        if instruction[0] == "A":
            definitive_list.insert(int(instruction[2]), instruction[1])
        elif instruction[0] == "R":
            definitive_list.remove(instruction[1])
        elif instruction[0] == "P":
            print(definitive_list)

if __name__ == "__main__":
    main()

File: atividade-02/lua/arq.lua
local function readFile(fileName)
    local file = io.open(fileName, "r")
    local lines = {}

    -- Lê todas as linhas do arquivo e as adiciona na tabela lines
    for line in file:lines() do
        table.insert(lines, line)
    end

    file:close()

    -- Remove a última linha se ela for uma string vazia
    if lines[#lines] == "" then
        table.remove(lines)
    end

    return lines
end

local function split(str, sep)
    local result = {}
    for match in (str .. sep):gmatch("(.-)" .. sep) do
        table.insert(result, match)
    end
    return result
end

local lines = readFile("arq-novo.txt")

-- Separa a primeira linha em um array usando o split
local definitiveList = split(lines[1], " ")
table.remove(lines, 1)

local instructionList = {}

-- Cria uma lista de instruções separadas
for i = 1, #lines do
    local instructions = split(lines[i], " ")
    table.insert(instructionList, instructions)
end

-- Processa cada instrução
for i = 1, #instructionList do
    local instruction = instructionList[i]

    if instruction[1] == "A" then
        table.insert(definitiveList, tonumber(instruction[3]) + 1, instruction[2])
    end

    if instruction[1] == "R" then
        for j = #definitiveList, 1, -1 do
            if definitiveList[j] == instruction[2] then
                table.remove(definitiveList, j)
                break
            end
        end
    end

    if instruction[1] == "P" then
        for _, value in ipairs(definitiveList) do
            io.write(value .. " ")
        end
        print()
    end
end
